tests/helpers/acceptance.js

```diff
@@ -8,6 +8,12 @@ var Promise           = require('../../lib/ext/promise');
 var root = path.resolve(__dirname, '..', '..');
 
 var PackageCache = require('../../tests/helpers/package-cache');
+var CommandGenerator = require('../../tests/helpers/command-generator');
+
+// We must link `ember-cli`.
+var npm = new CommandGenerator(require.resolve('npm/bin/npm-cli.js'));
+npm.invoke('link', { cwd: root });
```

Comment:
This discussion has been concluded: implemented a custom manual linking story inside of `PackageCache` which doesn't aim to imitate the package managers, just get "close enough" with running linking. Consumer is still responsible for making all things work.

Barring needing a rebase, here's the implementation: https://github.com/ember-cli/ember-cli/pull/6576/commits/e8d1ce0432095f54e543fa907f1c3460073ebc9b